### DISCUSSION TITLE
About the Direct3D 12 rendering support

### DISCUSSION THREAD

**Discussion by Viciouss** [godotengine/godot-proposals#9082] [Engine Core]
Hello there,

I just went through the blog post for the dev 3 snapshot, and I noticed the problem with dxil.dll. Just yesterday I happend to read another dev log of the mach engine where the author describes his share of problems regarding the DirectX shader compiler and a possible solution with no dependency to dxil.dll at all. I figured that this might help, maybe it's also already known or this doesn't work with Godot for some reason, I just thought that it couldn't hurt posting about it because I couldn't find it mentioned anywhere. And it's a rather new blog post as well.

Here is the link: [https://devlog.hexops.com/2024/building-the-directx-shader-compiler-better-than-microsoft/](https://devlog.hexops.com/2024/building-the-directx-shader-compiler-better-than-microsoft/)

Disclaimer: I have little to no experience with C++, Direct3D, dlls, shaders and whatnot. I just read a lot of articles in the internet that seem to be interesting even if I don't fully understand them.

Kind Regards
Martin

**Comment by Zireael07** [comment]
Two days ago there was a PR with this solution, unfortunately Bastiaan closed it

**Reply by Calinou** [reply]
This is the pull request in question: https://github.com/godotengine/godot/pull/88175

As the comment says, it would be legally risky to incorporate this. This can be compared to Dolphin bundling Wii encryption keys (which has *de facto* prevented them from releasing on Steam).

We'll have to figure out a different solution such as downloading the official `dxil.dll` on demand from the project manager or editor, like it's done for FBX2glTF.

**Reply by Viciouss** [reply]
That was a quick pull request indeed. I guess it's difficult to get in touch with Microsoft to clarify whether they would allow using the implementation.

**Reply by theraot** [reply]
Is this really an issue of asking for clarification from Microsoft?
